app/controllers/util/DisplayPropertiesUtils.scala

```diff
@@ -55,6 +55,38 @@ class DisplayPropertiesUtils(displayProperties: List[DisplayProperty], customMet
           InputNameAndValue(property.propertyName, customMetadata.defaultValue),
           required
         )
+      case "small text" => generateTextField(property, customMetadata)
+      case "date" =>
+        DateField(
+          property.propertyName,
+          property.displayName,
+          property.description,
+          property.multiValue,
+          InputNameAndValue("Day", "", "DD"),
+          InputNameAndValue("Month", "", "MM"),
+          InputNameAndValue("Year", "", "YYYY"),
+          required
+        )
+      case "radial" =>
+        val defaultOption = if (customMetadata.defaultValue.toBoolean) "yes" else "no"
```

Comment:
There is a amount of logic for this field that justifies abstracting it out into its own function, eg `generateRadialField`. Same as you've done for the "small text" field